src/praisonai-agents/praisonaiagents/tools/lsp_tools.py

```
@@ -0,0 +1,461 @@
+def _run_lsp(language: str, coro_factory):
+    """Spawn an LSP client for *language*, open the doc, run *coro_factory*.
+
+    ``coro_factory`` is an ``async def(client) -> result``.  Handles start/stop
+    and never re-enters a running event loop (returns a clear message instead).
+    Returns ``(result, None)`` on success or ``(None, message)`` on failure.
+    """
+    import asyncio
+
+    try:
+        from ..lsp.client import LSPClient
+    except Exception as e:  # pragma: no cover - import guarded
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)